File: Leetcode/03_Hash-Table/018_0001_two-sum.py
# https://leetcode.cn/problems/two-sum/
# e

class Solution(object):
    def twoSum(self, nums, target):
        dct = dict()
        for i in range(len(nums)):
            if target - nums[i] in dct:
                return [i, dct[target-nums[i]]]
            dct[nums[i]] = i
        return []

File: Leetcode/03_Hash-Table/018_0001_two-sum.go
// https://leetcode.cn/problems/two-sum/
// e

package main

func twoSum(nums []int, target int) []int {
	dict := make(map[int]int)
	for i, v := range nums {
		if val, exists := dict[target-v]; exists {
			return []int{i, val}
		}
		dict[v] = i
	}
	return []int{}
}
